### PR TITLE
Fix handling of single quotes in `RedshiftToS3Operator`

### LINKED ISSUE #30287: RedshiftToS3 Operator Wrapping Query in Quotes Instead of $$
### Apache Airflow version

2.5.2

### What happened

When passing a select_query into the RedshiftToS3 Operator, the query will error out if it contains any single quotes because the body of the UNLOAD statement is being wrapped in single quotes. 

### What you think should happen instead

Instead, it's better practice to use the double dollar sign or dollar quoting to signify the start and end of the statement to run. This removes the need to escape any special characters and avoids the statement throwing an error in the common case of using single quotes to wrap string literals. 

### How to reproduce

Running the RedshiftToS3 Operator with the sql_query: `SELECT 'Single Quotes Break this Operator'` will throw the error

### Operating System

NAME="Amazon Linux" VERSION="2" ID="amzn" ID_LIKE="centos rhel fedora" VERSION_ID="2" PRETTY_NAME="Amazon Linux 2" ANSI_COLOR="0;33" CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2" HOME_URL="https://amazonlinux.com//"

### Versions of Apache Airflow Providers

apache-airflow[package-extra]==2.4.3
apache-airflow-providers-amazon

### Deployment

Amazon (AWS) MWAA

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by vincbeck**
Continue the work started in #32915.
Closes: #30287.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by cjames23 — APPROVED** [review]
Looks good to me!

File: airflow/providers/amazon/aws/transfers/redshift_to_s3.py [hunk 1]
```diff
@@ -141,8 +141,10 @@ def __init__(
     def _build_unload_query(
         self, credentials_block: str, select_query: str, s3_key: str, unload_options: str
     ) -> str:
+        # Un-escape already escaped queries
+        select_query = select_query.replace("''", "'")
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/transfers/redshift_to_s3.py]: Would this break empty string literals…?

Comment (vincbeck) [reply airflow/providers/amazon/aws/transfers/redshift_to_s3.py]: Ouh! That's a good catch!

Comment (vincbeck) [reply airflow/providers/amazon/aws/transfers/redshift_to_s3.py]: I addressed it in [7d06f1e259a695fc68e16e50e9562f514dbb757d](https://github.com/apache/airflow/pull/35986/commits/7d06f1e259a695fc68e16e50e9562f514dbb757d). I also added a unit test for this use case

**Review by eladkal — APPROVED** [review]

**Pull request merged**